protocol/daemons/pricefeed/client/sub_task_runner.go

```diff
@@ -87,8 +91,17 @@ func (s *SubTaskRunnerImpl) StartPriceUpdater(
 		select {
 		case <-ticker.C:
 			err := RunPriceUpdaterTaskLoop(ctx, exchangeToMarketPrices, priceFeedServiceClient, logger)
-			if err != nil {
-				panic(err)
+
+			if err == nil {
+				// Record update success for the daemon health check.
+				c.RecordUpdateSuccess(&libtime.TimeProviderImpl{})
```

Comment:
nit: Your creating and destroying the `TimeProviderImpl` on every call. Adding it as a member variable to `SubTaskRunnerImpl` would allow setting it for testing purposes in the future or if you take my suggestion below about taking a time then you could pass in `time.Now()`.